### PR TITLE
2990 - Fixed units in scaled plots

### PR DESCRIPTION
**Pull request by lucas-wilkins**
There were some issues with variable re-use that caused the wrong units to appear on some of the differently scaled plots. Fixes this.

### CODE REVIEW

**Review by rozyczko — CHANGES_REQUESTED** [review]
Actual fix is working well. However, there are additional things modified/changed which should be made a separate PR, since they don't have anything to do with the current issue.

File: src/sas/qtgui/Utilities/GuiUtils.py [hunk 1]
```diff
@@ -694,12 +694,14 @@ def saveAnyData(data, wildcard_dict=None):
...
     parent = None
+    options = QtWidgets.QFileDialog.DontUseNativeDialog
     # Query user for filename.
     filename_tuple = QtWidgets.QFileDialog.getSaveFileName(parent,
                                                            caption,
                                                            "",
                                                            filter,
-                                                           "")
+                                                           "",
+                                                           options)
```
[the remark this answers is not in the document]
Comment (lucas-wilkins) [reply src/sas/qtgui/Utilities/GuiUtils.py:704]: I'm pretty confused by where this came from, it wasn't me

File: src/sas/qtgui/Utilities/GuiUtils.py [hunk 2]
```diff
@@ -739,16 +741,15 @@ def __init__(self, parent=None):
...
-        return QtGui.QValidator.Acceptable, pos
 
-        #try:
-        #    Formula(str(input))
-        #    self._setStyleSheet("")
-        #    return QtGui.QValidator.Acceptable, pos
+        try:
+            Formula(str(input))
+            self._setStyleSheet("")
+            return QtGui.QValidator.Acceptable
```
[the remark this answers is not in the document]
Comment (lucas-wilkins) [reply src/sas/qtgui/Utilities/GuiUtils.py:748]: again, I didn't do this

File: src/sas/qtgui/Utilities/GuiUtils.py [hunk 3]
```diff
@@ -1161,9 +1162,6 @@ def jdefault(o):
         if isinstance(o, numbers.Real):
             return float(o)
 
-        if isinstance(o, DataRole):
-            return o.value
-
```
[the remark this answers is not in the document]
Comment (lucas-wilkins) [reply src/sas/qtgui/Utilities/GuiUtils.py:1166]: dunno

**Comment by lucas-wilkins** [comment]
Clearly something has gone very wrong with this commit. Need to make a new one.

File: src/sas/qtgui/Utilities/GuiUtils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (lucas-wilkins) [reply src/sas/qtgui/Utilities/GuiUtils.py]: restored to previous code

File: src/sas/qtgui/Utilities/GuiUtils.py [hunk 4]
```diff
@@ -739,16 +741,15 @@ def __init__(self, parent=None):
...
+            Formula(str(input))
+            self._setStyleSheet("")
+            return QtGui.QValidator.Acceptable
 
-        #except Exception as e:
-        #    self._setStyleSheet("background-color:pink;")
-        #    return QtGui.QValidator.Intermediate, pos
+        except Exception as e:
+            self._setStyleSheet("background-color:pink;")
+            return QtGui.QValidator.Intermediate
```
[the remark this answers is not in the document]
Comment (lucas-wilkins) [reply src/sas/qtgui/Utilities/GuiUtils.py]: restored to previous code

File: src/sas/qtgui/Utilities/GuiUtils.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (lucas-wilkins) [reply src/sas/qtgui/Utilities/GuiUtils.py]: restored to previous code

**Comment by lucas-wilkins** [comment]
@rozyczko Could you check that this is now correct?

**Review by rozyczko — APPROVED** [review]
looks good now!

**Pull request merged**